aea/protocols/dialogue/base.py

```diff
@@ -948,6 +995,220 @@ def add_dialogue_endstate(
+        """
+        Get dialogue stored by it's label.
+
+        :param dialogue_label: label of the dialogue
+        :return: dialogue if presents or None
+        """
+        return self.dialogues.get(dialogue_label, None)
+
+    @property
+    def dialogues(self) -> Dict[DialogueLabel, Dialogue]:
+        """Get dictionary of dialogues in which the agent engages."""
+        return self._dialogues_by_dialogue_label
+
+    def get_dialogues_with_counterparty(self, counterparty: Address) -> List[Dialogue]:
+        """
+        Get the dialogues by address.
+
+        :param counterparty: the counterparty
+        :return: The dialogues with the counterparty.
+        """
+        return self._dialogue_by_address.get(counterparty, [])
+
+    def is_in_incomplete(self, dialogue_label: DialogueLabel) -> bool:
+        """Check dialogue label presents in list of incomplete."""
+        return dialogue_label in self._incomplete_to_complete_dialogue_labels
+
+    def set_incomplete_dialogue(
+        self, incomplete_dialogue_label, complete_dialogue_label
+    ) -> None:
+        """Set incomplete dialogue label."""
+        self._incomplete_to_complete_dialogue_labels[
+            incomplete_dialogue_label
+        ] = complete_dialogue_label
+
+    def is_dialogue_present(self, dialogue_label: DialogueLabel) -> bool:
+        """Check dialogue with label specified presents in storage."""
+        return dialogue_label in self._dialogues_by_dialogue_label
+
+    def get_latest_label(self, dialogue_label: DialogueLabel) -> DialogueLabel:
+        """Get latest label for dialogue."""
+        return self._incomplete_to_complete_dialogue_labels.get(
+            dialogue_label, dialogue_label
+        )
+
+
+class PersistDialoguesStorage(BasicDialoguesStorage):
+    """
+    Persist dialogues storage.
+
+    Uses generic storage to load/save dialogues data on setup/teardown.
+    """
+
+    INCOMPLETE_DIALOGUES_OBJECT_NAME = "incomplete_dialogues"
+
+    def __init__(self, dialogues: "Dialogues") -> None:
+        """Init dialogues storage."""
+        super().__init__(dialogues)
+
+        self._skill_component: Optional[SkillComponent] = self.get_skill_component()
+
+    @staticmethod
+    def get_skill_component() -> Optional[SkillComponent]:
+        """Get skill component dialogues storage constructed for."""
+        caller_object = find_caller_object(SkillComponent)
+        if not caller_object:  # pragma: nocover
+            return None
+        return caller_object
+
+    def _get_collection_name(self) -> Optional[str]:
+        """Generate collection name based on the dialogues class name and skill component."""
+        if not self._skill_component:  # pragma: nocover
+            return None
+        return "_".join(
+            [
+                self._skill_component.skill_id.author,
+                self._skill_component.skill_id.name,
+                self._skill_component.name,
```

Comment:
also skill component type should feature here to be sure?

Reply:
added. nice point